src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -33,6 +34,9 @@ public AuthorizationMiddleware(RequestDelegate next, IAuthorizationPolicyProvide
     {
         _next = next ?? throw new ArgumentNullException(nameof(next));
         _policyProvider = policyProvider ?? throw new ArgumentNullException(nameof(policyProvider));
+
+        // Only try to cache combined policies for the default policy provider
+        _cacheCombinedPolicy = _policyProvider is DefaultAuthorizationPolicyProvider;
```

Comment:
```suggestion
        _cacheCombinedPolicy = _policyProvider.GetType() == typeof(DefaultAuthorizationPolicyProvider);
```

We don't want derived types either, `GetPolicyAsync` is virtual.

Reply:
This makes it very hard (impossible?) at least with mock, I've tried (and failed) a few different ways of trying to test this but its hard when I can't derive or Mock the policyProvider I want to inspect.  Are we okay just testing the non cached cases?

Reply:
You could set an internal bool that tests override.

Reply:
Oh i get it, so test the caching behavior itself, just not against the DefaultAuthorizationPolicyProvider, good idea